staging/src/k8s.io/apiextensions-apiserver/pkg/registry/customresource/etcd.go

```diff
@@ -237,57 +237,76 @@ func (r *ScaleREST) Get(ctx context.Context, name string, options *metav1.GetOpt
 }
 
 func (r *ScaleREST) Update(ctx context.Context, name string, objInfo rest.UpdatedObjectInfo, createValidation rest.ValidateObjectFunc, updateValidation rest.ValidateObjectUpdateFunc, forceAllowCreate bool, options *metav1.UpdateOptions) (runtime.Object, bool, error) {
```

Comment:
this is in the right direction, but prefer using the existing clientretry.OnError helper:

```go
retryOnConflict := false
isRetriableConflict := func(err error) bool {
  return apierrors.IsConflict(err) && retryOnConflict
}

var (
  newScale runtime.Object
  err error
)

err = clientretry.OnError(clientretry.DefaultRetry, isRetriableConflict, func() error {
  obj, err := r.store.Get(ctx, name, &metav1.GetOptions{})
  ...
 
  if len(scale.ResourceVersion) == 0 {
    // no resourceVersion was provided, so use the existing cr resourceVersion and retry conflicts server-side
    retryOnConflict = true
  } else {
    // a resourceVersion was provided, so use it and return conflict errors to the user
    retryOnConflict = false
    cr.SetResourceVersion(scale.ResourceVersion)
  }
  ...
  newScale, _, err = scaleFromCustomResource(cr, r.specReplicasPath, r.statusReplicasPath, r.labelSelectorPath)
  if err != nil {
    return apierrors.NewBadRequest(err.Error())
  }
  return nil
})

return newScale, false, err
```

Reply:
@liggitt Thanks for your detailed guidance! I have updated the code.